base/regex.jl

```diff
@@ -418,11 +418,26 @@ function match(re::Regex, str::Union{SubString{String}, String}, idx::Integer,
                                         SubString(str, unsafe_load(p,2i+1)+1,
                                                   prevind(str, unsafe_load(p,2i+2)+1)) for i=1:n]
     off = Int[ unsafe_load(p,2i+1)+1 for i=1:n ]
-    result = RegexMatch(mat, cap, unsafe_load(p,1)+1, off, re)
+    result = RegexMatch{String}(mat, cap, unsafe_load(p,1)+1, off, re)
```

Comment:
Without it I see (on `make`):
```
error during bootstrap:
LoadError("sysimg.jl", 3, LoadError("Base.jl", 314, LoadError("version.jl", 168, LoadError("version.jl", 168, MethodError(LoadError(at "sysimg.jl" line 3: LoadError(at "Base.jl" line 314: LoadError(at "version.jl" line 168: LoadError(at "version.jl" line 168: MethodError(f=Base.RegexMatch{S} where S<:AbstractString, args=(Base.SubString{String}(string="0-", offset=0, ncodeunits=2), Array{Union{Nothing, Base.SubString{String}}, (7,)}[
  Base.SubString{String}(string="0-", offset=0, ncodeunits=1),
  nothing,
  nothing,
  Base.SubString{String}(string="0-", offset=1, ncodeunits=1),
  nothing,
  nothing,
  nothing], 0x0000000000000001, Array{Int64, (7,)}[1, 0, 0, 2, 0, 0, 0], Base.Regex(pattern="^
    v?                                      # prefix        (optional)
    (\d+)                                   # major         (required)
    (?:\.(\d+))?                            # minor         (optional)
    (?:\.(\d+))?                            # patch         (optional)
    (?:(-)|                                 # pre-release   (optional)
    ([a-z][0-9a-z-]*(?:\.[0-9a-z-]+)*|-(?:[0-9a-z-]+\.)*[0-9a-z-]+)?
    (?:(\+)|
    (?:\+((?:[0-9a-z-]+\.)*[0-9a-z-]+))?    # build         (optional)
    ))
$", compile_options=0x040a008a, match_options=0x40000000, regex=0x00000000040c4d00)), world=0x00000000000035ba)))))
jl_method_error_bare at /home/tec/Desktop/TEC/Projects/available/julia/src/gf.c:2203
jl_method_error at /home/tec/Desktop/TEC/Projects/available/julia/src/gf.c:2221
jl_lookup_generic_ at /home/tec/Desktop/TEC/Projects/available/julia/src/gf.c:3050 [inlined]
ijl_apply_generic at /home/tec/Desktop/TEC/Projects/available/julia/src/gf.c:3065
match at ./regex.jl:421
```

the good news is I can't see any instances of `RegexMatch(...)` in the wild outside of Julia forks: https://grep.app/search?q=RegexMatch%28&filter[lang][0]=Julia

Reply:
Ok, this seems to be solved by adding this constructor:
```julia
RegexMatch(match::SubString{S}, captures::Vector{Union{Nothing,SubString{S}}},
           offset::Union{Int, UInt}, offsets::Vector{Int}, regex::Regex) where {S<:AbstractString} =
    RegexMatch{S}(match, captures, offset, offsets, regex)
```

Yes, the `Union{Int, UInt}` is needed (because of `unsafe_load(p,1)+1` on line 425)